### DISCUSSION TITLE
Export labels as image

### DISCUSSION THREAD

**Discussion by shubham-scisar** [wkentaro/labelme#1100] [Q&A / Help] [answered]
Is there any functionality to export the final image as **mask image** having different classes. Like we are using the labels to be exported as json file. Instead of json file, it should be exported as a single image having different labels. Kindly guide.

**Comment by Liu-ZhenJi** [comment]
The script below are the workable codes to translate the json file to mask image. Wish this is helpful for you. : )


def json2seg_label(json_root_dir, image_root_dir, save_dir):

    for file_name in os.listdir(json_root_dir):
        print('The current json_name: ', file_name)

        label_dict = parse_json(os.path.join(json_root_dir, file_name))

        image_tmp = cv2.imread(os.path.join(image_root_dir, file_name.replace('.json', '.png')))
        mask_1 = np.zeros(image_tmp[:, :, 0].shape, np.uint8)
        mask_3 = np.zeros(image_tmp[:, :, 0].shape, np.uint8)

        cv2.fillPoly(mask_1, label_dict['1'], 1)
        cv2.fillPoly(mask_3, label_dict['3'], 3)

        mask_4 = copy.deepcopy(mask_3)
        mask_4[mask_1==1] = 1

        cv2.imwrite(os.path.join(save_dir, file_name.replace('.json', '.png')), mask_4)


def parse_json(json_dir):

    with open(json_dir, 'r') as f:
        label_json = json.load(f)

    label_dict = {}
    for i in range(len(label_json['shapes'])):

        temp_label = label_json['shapes'][i]['label']
        # if temp_label == '3':
        #     continue
        if temp_label not in label_dict.keys():
            label_dict[temp_label] = []

        temp_list = label_json['shapes'][i]['points']
        tmp_label_list = []
        for pair in temp_list:
            # print('The pair: ', pair)
            tmp_label_list.append([round(pair[0]), round(pair[1])])  # The type must be integer.
        tmp_label_array = np.array([tmp_label_list])
        label_dict[temp_label].append(tmp_label_array)

    return label_dict

**Comment by wkentaro** [comment, marked as answer]
`labelme2voc.py` saves labels as PNG file.
